src/components/app.tsx

```diff
@@ -55,6 +55,9 @@ class App extends React.Component<Props & { match: any; location: any; showExamp
     );
 
     const parameter = this.props.match.params;
+    if (parameter.mode) {
+      this.props.setModeOnly(parameter.mode);
+    }
     this.setSpecInUrl(parameter);
```

Comment:
Doesn't this call `setState` twice? Is that a problem?

Reply:
The first one will just update the mode, you can check the reducer, it won't cause a problem.